src/wasm-builder.h

```diff
@@ -310,8 +310,9 @@ class Builder {
 
   static Index addParam(Function* func, Name name, WasmType type) {
     // only ok to add a param if no vars, otherwise indices are invalidated
-    assert(func->localIndices.size() == func->params.size());
     assert(name.is());
+    assert(func->localIndices.find(name) == func->localIndices.end());
+    assert(func->localIndices.size() == func->params.size());
```

Comment:
Sorry, but the more I think about this, I have an uneasy feeling. Something in the design (way back) may have been wrong here.

How about changing `localNames` from a vector to a map (like `localIndices`)? Then we wouldn't require names to exist, or have these asserts, etc. It would then be fine for some params and some vars to have names while others do not, which seems the simplest design (i.e. no different treatment of params vs vars). In that case we also don't need the loop in addVar, since we don't need to add a name.